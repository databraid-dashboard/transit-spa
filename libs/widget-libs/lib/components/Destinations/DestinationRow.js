var _jsxFileName = '../../src/components/Destinations/DestinationRow.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { Container, Segment } from 'semantic-ui-react';

import JourneyTable from '../Journeys/JourneyTable';
import DestinationHeader from './Destination/DestinationHeader';
import './destination.css';

var DestinationRow = function DestinationRow(_ref) {
  var id = _ref.id,
      name = _ref.name;
  return React.createElement(
    Container,
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 11
      },
      __self: _this
    },
    React.createElement(
      Segment,
      { compact: true, __source: {
          fileName: _jsxFileName,
          lineNumber: 12
        },
        __self: _this
      },
      React.createElement(DestinationHeader, { name: name, __source: {
          fileName: _jsxFileName,
          lineNumber: 13
        },
        __self: _this
      })
    ),
    React.createElement(
      Segment,
      { compact: true, textAlign: 'center', __source: {
          fileName: _jsxFileName,
          lineNumber: 15
        },
        __self: _this
      },
      React.createElement(JourneyTable, { id: id, __source: {
          fileName: _jsxFileName,
          lineNumber: 16
        },
        __self: _this
      })
    )
  );
};

DestinationRow.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
};

DestinationRow.defaultProps = {
  id: 1,
  name: ''
};

export default DestinationRow;