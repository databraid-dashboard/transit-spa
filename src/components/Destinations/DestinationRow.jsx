import React from 'react';
import PropTypes from 'prop-types';

import { Segment } from 'semantic-ui-react';

import JourneyTable from '../Journeys/JourneyTable';
import DestinationHeader from './Destination/DestinationHeader';
import RemoveDestinationButton from './RemoveDestinationButton';
import './destination.css';

const DestinationRow = ({ id, name }) => (
  <div>
    <RemoveDestinationButton id={id} />
    <Segment compact>
      <DestinationHeader name={name} />
    </Segment>
    <Segment textAlign="center" compact>
      <JourneyTable id={id} />
    </Segment>
  </div>
);

DestinationRow.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

DestinationRow.defaultProps = {
  id: 1,
  name: 'Civic Center',
};

export default DestinationRow;
