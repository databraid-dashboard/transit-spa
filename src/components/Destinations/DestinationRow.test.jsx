import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DestinationRow from './DestinationRow';

describe('DestinationRow', () => {
  it('should render when given a name and id', () => {
    const id = 1;
    const name = 'Tartine';
    const component = shallow(<DestinationRow id={id} name={name} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
