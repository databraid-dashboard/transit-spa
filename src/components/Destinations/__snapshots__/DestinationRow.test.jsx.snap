// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationRow should render an empty form 1`] = `
<Container>
  <Segment
    class="destination-row"
    compact={true}
  >
    <DestinationHeader
      name=""
    />
  </Segment>
  <Segment
    class="destination-row"
    compact={true}
    textAlign="center"
  >
    <Wrapper
      id={1}
    />
  </Segment>
</Container>
`;
